sklearn/preprocessing/_discretization.py

```diff
@@ -138,6 +144,15 @@ def fit(self, X, y=None):
         self
         """
         X = check_array(X, dtype='numeric')
+        if self.dtype in (np.float64, np.float32):
```

Comment:
would it be a good idea to store this tuple of types somewhere such that it's reusable? it seems to be used in multiple places and we don't want to have a case where it's updated in one location and not in others...

Reply:
@rth suggested to explicitly write `(np.float64, np.float32)` for a `check_array` function, so I use it everywhere. I don't have any preference for this point.